Fix passive homes

julie-vienne opened the pull request:
Issue #28

julie-vienne committed:
Fix replace_other_prm

julie-vienne commented:
@floracharbo your comment did solve this specific problem, thanks.

However, I've noticed another problem if `n_homes = 2, n_homesP = 3`:

```
self.max_charge_t = np.where(last_step and self.avail_car, self.store0, self.cap)
```
as `self.store0` has the dimension of `n_homes` and `self.cap` has the dimension of `n_homesP`.

If they're the same, e.g. both 2, then comes another problem with the following assertion:

`assert abs(reward + res['hourly_total_costs'][time_step]) < 5e-3`

I thought it would be nice to directly include the passive houses in the reactive power PR but I will write my equations to set the reactive power of passive houses to zero until this issue is solved.

floracharbo commented:
> harge_t = np.where(last_step and self.avail_car, self.store0, self.cap)

yes, we need to switch using the `set_passive_active` method when using the Battery methods to analyse the cars of the active of passive houses.

floracharbo committed:
deep copy of prm for car as it is edited

floracharbo committed:
set active setting for active steps

floracharbo committed:
remove obsolete methods and variables

floracharbo committed:
add sum_netp0_abs to hourly_distribution_network_export_costs in post_processing opt results

also:
- self.loads['netp0'] list to np for conciseness and efficiency
- tidy/conciseness
- rename variables for clarity

floracharbo committed:
method move to env to avoid repetitions + self.prm["loads"] passive arrays

floracharbo committed:
tidy result of passive steps in self.loads

floracharbo committed:
minor tidy/lint

floracharbo committed:
add passive costs to hourly battery costs in post_processing optimisation

floracharbo committed:
only sum hourly export costs over current time step (postprocessing)

floracharbo commented on src/simulations/hedge.py:
```diff
@@ -40,29 +41,31 @@ def __init__(
-    def _replace_other_prm(self, prm, other_prm):
-        prm = prm.copy()
+    def _replace_car_prm(self, prm, other_prm):
+        prm = copy.deepcopy(prm)
```
@julie-vienne fyi this was the main problem, that there there was a shallow copy instead of a deep copy, so when we edited the car info in the `passive_hedge` this also edited the main `prm`

julie-vienne replied:
I see, good to know!

floracharbo reviewed (commented):
changes reviewed, now we just need to solve conflicts with main

floracharbo committed:
Merge branch 'main' of https://github.com/floracharbo/MARL_local_electricity into fix_passive_homes_voltages

floracharbo commented:
note that I have debugged the passive homes with `manage_voltage = False`. I will merge at this point but I will leave it to you to implement it with the network modelling @julie-vienne.

julie-vienne commented:
Issue #35

julie-vienne commented:
@floracharbo would you prefer that I open a new PR to work on #35?

floracharbo commented:
> @floracharbo would you prefer that I open a new PR to work on #35?

I don't mind, if you reuse that branch please just wait until I have merged #31 (this may take a bit of time as I am waiting for tests to pass) and merge main into the branch again, so you are not too behind on the changes. Or it might just be easier to create a new branch from main.

julie-vienne commented:
@floracharbo would it be okay for you if I tackle issue #35 with the reactive power issue as they are closely related?

julie-vienne commented:
@floracharbo I think we can actually close this issue as we've seen in #29 that the passive homes now work when `manage_voltage = False` and `manage_agg_power = True`, I think we can solve it for the voltage management directly when we add the reactive power.

floracharbo commented:
Yes, this is already closed. And I leave you to tackle #35 with the power flow modelling.
